adr/configuration.py

```diff
@@ -70,27 +70,34 @@ class Configuration(Mapping):
         "url": "https://activedata.allizom.org/query",
         "verbose": False,
     }
-    locked = False
 
     def __init__(self, path=None):
         self.path = Path(path or os.environ.get("ADR_CONFIG_PATH") or self.DEFAULT_CONFIG_PATH)
-
-        self._config = self.DEFAULTS.copy()
+        self._config={}
+        self.update(self.DEFAULTS)
         if self.path.is_file():
             with open(self.path, "r") as fh:
                 content = fh.read()
-                self.merge(parse(content)["adr"])
+                self.update(parse(content)["adr"])
         else:
             logger.warning(f"Configuration path {self.path} is not a file.")
 
+    def update(self, config):
+        """
+        Update the configuration object with new parameters
+        :param config: dict of configuration
+        """
+        for k, v in config.items():
+            if v != None:
+                self._config[k] = v
```

Comment:
Should you use the `merge` function you removed below instead? It will append to lists and update dicts rather than overwrite them.  There are some `DEFAULTS` (namely `sources`) that should always be applied.